fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/RecordQuery.java

```diff
@@ -139,7 +139,7 @@ public List<KeyExpression> getRequiredResults() {
      */
     public void validate(@Nonnull RecordMetaData metaData) {
         for (String recordTypeName : recordTypes) {
-            final RecordType recordType = metaData.getRecordType(recordTypeName);
+            final RecordType recordType = metaData.getIndexableRecordType(recordTypeName);
```

Comment:
I would somewhat rather introduce a new `getQueryableRecordType` method or some name like that for this case, even if the present definition would be the same as `getIndexableRecordType`, that is, physical or synthetic. This isn't valid because it's indexable, really, but because this PR fixes support for planning it.

Reply:
done.